File: aurafit-frontend/src/Models/DTOS/ProductVariantCreateDTO.ts
import {Size} from "../Size.ts";
import {Color} from "../Color.ts";
import {Material} from "../Material.ts";
import {Product} from "../Product.ts";



export class ProductVariantCreateDTO {
    size: Size
    color: Color
    material: Material
    basePrice: number
    stockQuantity: number
    product: Product

    constructor(size: Size, color: Color, material: Material, basePrice: number, stockQuantity: number, product: Product) {
        this.size = size
        this.color = color
        this.material = material
        this.basePrice = basePrice
        this.stockQuantity = stockQuantity
        this.product = product
    }



}

File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/productVariant/ProductVariantCreateDto.java
package app.aurafitbackend.productVariant;

import app.aurafitbackend.color.Color;
import app.aurafitbackend.material.Material;
import app.aurafitbackend.product.Product;
import app.aurafitbackend.size.Size;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductVariantCreateDto {

    private Size size;

    private Color color;


    private Material material;

    private BigDecimal basePrice;
    private BigDecimal salePrice;

    private Integer stockQuantity;

    private Product product;
}
